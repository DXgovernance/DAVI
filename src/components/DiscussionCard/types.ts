import { Discussion } from 'components/Forum/types';

export interface DiscussionCardProps {
  discussion: Discussion;
}
