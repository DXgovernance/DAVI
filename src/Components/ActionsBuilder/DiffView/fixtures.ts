export const mockCodeOriginal = `
  {
    "colors": {
      "text": "red",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "lightGrey": "#BDC0C7",
        "grey": "orange"
      },
    }
  }
`;

export const mockCodeUpdated = `
  {
    "colors": {
      "text": "#fff",
      "primary": "#295FF4",
      "secondary": "#181B1B",
      "red": "#E75C5C",
      "orange": "#D17A06",
      "proposalText": {
        "grey": "#A1A6B0",
        "lightGrey": "#BDC0C7"
      },
    }
  }
`;
