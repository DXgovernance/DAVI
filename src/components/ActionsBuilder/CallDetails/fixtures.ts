import {
  ApproveSendTokens,
  DecodedCall,
  SupportedAction,
} from 'components/ActionsBuilder/types';
import ERC20 from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20.abi);

export const emptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const decodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0x95a223299319022a842d0dfe4851c145a2f615b9',
  value: BigNumber.from(0),
  args: {
    _to: '0x0000000000000000000000000000000000000000',
    _value: BigNumber.from(1000),
  },
};

export const approvetokenSpendingMock: ApproveSendTokens = {
  callType: SupportedAction.GENERIC_CALL,
  from: '0x0000000000000000000000000000000000000000',
  to: '0x0000000000000000000000000000000000000000',
  value: BigNumber.from(0),
  function: null,
  args: {},
  amount: BigNumber.from(1000),
  token: '0x3f943f38b2fbe1ee5daf0516cecfe4e0f8734351',
};
