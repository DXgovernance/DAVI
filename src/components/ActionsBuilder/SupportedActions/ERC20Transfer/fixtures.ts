import { DecodedCall, SupportedAction } from 'components/ActionsBuilder/types';
import ERC20ABI from 'contracts/ERC20.json';
import { BigNumber, utils } from 'ethers';

const ERC20Contract = new utils.Interface(ERC20ABI.abi);

export const erc20TransferEmptyDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '',
  value: BigNumber.from(0),
  args: {
    _to: '',
    _value: BigNumber.from(0),
  },
};

export const erc20TransferDecodedCallMock: DecodedCall = {
  from: '0x0000000000000000000000000000000000000000',
  callType: SupportedAction.ERC20_TRANSFER,
  function: ERC20Contract.getFunction('transfer'),
  to: '0xfFb1cd0F95368DDd06D556161c5D3d9f0f4Fe6d2',
  value: BigNumber.from(0),
  args: {
    _to: '0xaF8eB8C3A5d9d900AA0B98e3Df0bcC17d3C5F698',
    _value: BigNumber.from('5000000000000000000'),
  },
};
