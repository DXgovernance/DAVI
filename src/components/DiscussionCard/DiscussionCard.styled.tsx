import { CardWrapper } from 'components/Card';
import { Flex } from 'components/primitives/Layout';
import { Heading } from 'components/primitives/Typography';
import styled from 'styled-components';

export const CardTitle = styled(Heading)`
  font-size: 1rem;
  font-weight: 700;
  margin: 0;
  margin-bottom: 12px;
  @media only screen and (min-width: 768px) {
    font-size: 1.25rem;
  }
`;

export const FooterElement = styled(Flex)`
  flex-direction: row;
  column-gap: 10px;
  margin-right: 18px;
  font-weight: 600;
`;

export const MainWrapper = styled(CardWrapper)<{
  disabled?: boolean;
}>`
  margin-bottom: 1rem;
  padding: 20px 24px;
  color: ${({ theme }) => theme.colors.grey2};

  &:hover {
    ${({ theme, disabled }) =>
      disabled
        ? `cursor: default;`
        : `border-color: ${theme.colors.border3};
          color: ${theme.colors.text};`}
  }
`;
