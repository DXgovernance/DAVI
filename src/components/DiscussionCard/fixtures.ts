import { DiscussionCardProps } from './types';

export const fullParameters: DiscussionCardProps = {
  discussion: {
    stream_id:
      'kjzl6cwe1jw148wi063e3zuf4o1cgvcx9j40kps2fdihdfu9t5yqyntgfq4xsl1',
    type: null,
    content: {
      body: 'Body of the discussion',
      data: {},
      title: 'Title of the discussion',
      master: null,
      context: '0xCC88B92dcBA3003d717050dC67dDce4CBD1a3CE4',
      replyTo: null,
      mentions: [],
    },
    context: '0xCC88B92dcBA3003d717050dC67dDce4CBD1a3CE4',
    creator: 'did:pkh:eip155:31337:0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
    creator_details: {
      did: 'did:pkh:eip155:31337:0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
      profile: null,
      metadata: {
        chain: 'eip155:31337',
        address: '0xc5b20ade9c9cd5e0cc087c62b26b815a4bc1881f',
        ensName: null,
      },
    },
    group_id: null,
    context_details: {
      group_id: null,
      group_details: null,
      channel_id: null,
      channel_details: null,
    },
    master: null,
    reply_to: null,
    reply_to_details: null,
    count_likes: 7,
    count_haha: 0,
    count_downvotes: 0,
    count_replies: 3,
    timestamp: 1661975585,
  },
};
