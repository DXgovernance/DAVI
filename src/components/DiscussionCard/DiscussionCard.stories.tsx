import { ComponentStory, ComponentMeta } from '@storybook/react';
import { fullParameters } from './fixtures';
import { DiscussionCard } from './DiscussionCard';

export default {
  title: 'DiscussionCard',
  component: DiscussionCard,
  argTypes: {
    discussion: {
      description: 'Object containing data about a single discussion',
    },
  },
} as ComponentMeta<typeof DiscussionCard>;

const Template: ComponentStory<typeof DiscussionCard> = args => (
  <DiscussionCard {...args} />
);

export const Simple = Template.bind({});
Simple.args = fullParameters;
