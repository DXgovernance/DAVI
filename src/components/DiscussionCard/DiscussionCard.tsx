import { Flex } from 'components/primitives/Layout';
import { UnstyledLink } from 'components/primitives/Links';
import { shortenAddress } from 'utils';
import { CardTitle, MainWrapper, FooterElement } from './DiscussionCard.styled';
import { BiHeart } from 'react-icons/bi';
import { MdReply } from 'react-icons/md';
import useENSAvatar from 'hooks/Guilds/ens/useENSAvatar';
import { DiscussionCardProps } from './types';
import { useTypedParams } from 'Modules/Guilds/Hooks/useTypedParams';

export const DiscussionCard: React.FC<DiscussionCardProps> = ({
  discussion,
}) => {
  const { guildId, chainName } = useTypedParams();
  const creatorAddress = discussion.creator_details.metadata?.address;
  const { ensName } = useENSAvatar(creatorAddress, 1);

  return (
    <UnstyledLink
      to={`/${chainName}/${guildId}/discussion/${discussion.stream_id}`}
      data-testid="discussion-card"
    >
      <MainWrapper>
        <CardTitle>{discussion.content?.title}</CardTitle>
        <Flex direction="row" justifyContent="flex-start">
          <FooterElement>
            {ensName || shortenAddress(creatorAddress)}
          </FooterElement>
          <FooterElement>
            <BiHeart size="20px" /> {discussion?.count_likes}
          </FooterElement>
          <FooterElement>
            <MdReply size="20px" /> {discussion?.count_replies}
          </FooterElement>
        </Flex>
      </MainWrapper>
    </UnstyledLink>
  );
};
