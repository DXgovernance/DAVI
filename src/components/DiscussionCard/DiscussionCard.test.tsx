import { render } from 'utils/tests';
import '@testing-library/jest-dom';
import { DiscussionCard } from './DiscussionCard';
import { fullParameters } from './fixtures';

jest.mock('hooks/Guilds/ens/useENSAvatar', () => ({
  __esModule: true,
  default: () => ({
    ensName: 'test.eth',
  }),
}));

jest.mock('contexts/Guilds/orbis', () => ({
  useOrbisContext: () => ({
    orbis: {
      getPosts: () => ({
        data: [1, 2, 3],
      }),
    },
  }),
}));

jest.mock('@orbisclub/orbis-sdk', () => {});

jest.mock('wagmi', () => ({
  useNetwork: () => ({ chain: { id: 1 } }),
  chain: {
    mainnet: {},
  },
}));

describe('DiscussionCard', () => {
  it('should render with full parameters', async () => {
    const { container } = render(<DiscussionCard {...fullParameters} />);
    expect(container).toMatchSnapshot();
  });
});
