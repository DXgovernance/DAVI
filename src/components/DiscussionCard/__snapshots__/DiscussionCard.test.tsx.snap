// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiscussionCard should render with full parameters 1`] = `
.c1 {
  box-sizing: 'border-box';
  min-width: 0;
  margin: 0;
  padding: 0;
}

.c6 {
  display: Flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  border-radius: 10px;
  margin: 0;
  padding: 0;
}

.c7 {
  display: Flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  border-radius: 10px;
  margin: 0;
  padding: 0;
}

.c0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fff;
}

.c0:focus,
.c0:hover,
.c0:visited,
.c0:link,
.c0:active {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  border: 1px solid #303338;
  border-radius: 10px;
}

.c4 {
  font-family: 'Inter',-apple-system,BlinkMacSystemFont,'Segoe UI','Roboto','Oxygen','Ubuntu','Cantarell','Fira Sans','Droid Sans','Helvetica Neue',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 500;
}

.c4 strong,
.c4 b {
  font-weight: 600;
}

.c5 {
  font-size: 1rem;
  font-weight: 700;
  margin: 0;
  margin-bottom: 12px;
}

.c8 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-column-gap: 10px;
  column-gap: 10px;
  margin-right: 18px;
  font-weight: 600;
}

.c3 {
  margin-bottom: 1rem;
  padding: 20px 24px;
  color: #BDC0C7;
}

.c3:hover {
  border-color: #fff;
  color: #fff;
}

@media only screen and (min-width:768px) {
  .c5 {
    font-size: 1.25rem;
  }
}

<div>
   
  <a
    class="c0"
    data-testid="discussion-card"
    href="/undefined/undefined/discussion/kjzl6cwe1jw148wi063e3zuf4o1cgvcx9j40kps2fdihdfu9t5yqyntgfq4xsl1"
  >
    <div
      class="c1 c2 c3"
    >
      <h2
        class="c4 c5"
      >
        Title of the discussion
      </h2>
      <div
        class="c6"
        direction="row"
      >
        <div
          class="c7 c8"
        >
          test.eth
        </div>
        <div
          class="c7 c8"
        >
          <svg
            fill="currentColor"
            height="20px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 4.595a5.904 5.904 0 0 0-3.996-1.558 5.942 5.942 0 0 0-4.213 1.758c-2.353 2.363-2.352 6.059.002 8.412l7.332 7.332c.17.299.498.492.875.492a.99.99 0 0 0 .792-.409l7.415-7.415c2.354-2.354 2.354-6.049-.002-8.416a5.938 5.938 0 0 0-4.209-1.754A5.906 5.906 0 0 0 12 4.595zm6.791 1.61c1.563 1.571 1.564 4.025.002 5.588L12 18.586l-6.793-6.793c-1.562-1.563-1.561-4.017-.002-5.584.76-.756 1.754-1.172 2.799-1.172s2.035.416 2.789 1.17l.5.5a.999.999 0 0 0 1.414 0l.5-.5c1.512-1.509 4.074-1.505 5.584-.002z"
            />
          </svg>
           
          7
        </div>
        <div
          class="c7 c8"
        >
          <svg
            fill="currentColor"
            height="20px"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="20px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M10 9V5l-7 7 7 7v-4.1c5 0 8.5 1.6 11 5.1-1-5-4-10-11-11z"
            />
          </svg>
           
          3
        </div>
      </div>
    </div>
  </a>
   
</div>
`;
